### PR TITLE
chore(aws): Clean IAM resources

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

This PR makes no changes other than updating based on current best standards.  This is a cleanup before I overhaul some of the IAM resources

### CODE REVIEW

File: plugins/source/aws/resources/services/iam/saml_identity_providers.go [hunk 1]
```diff
@@ -32,3 +36,31 @@ func SamlIdentityProviders() *schema.Table {
...
+	p := resource.Item.(types.SAMLProviderListEntry)
+
+	providerResponse, err := svc.GetSAMLProvider(ctx, &iam.GetSAMLProviderInput{SAMLProviderArn: p.Arn})
+	if err != nil {
+		return err
+	}
+
+	resource.Item = models.IAMSAMLIdentityProviderWrapper{
+		GetSAMLProviderOutput: providerResponse,
+		Arn:                   *p.Arn,
```
Comment (candiduslynx) [inline plugins/source/aws/resources/services/iam/saml_identity_providers.go:62]: no real opinion, but maybe `aws.StringValue`?

File: plugins/source/aws/resources/services/iam/saml_identity_providers_fetch.go [hunk 2]
```diff
@@ -1,39 +1 @@
 package iam
-
```
Comment (candiduslynx) [inline plugins/source/aws/resources/services/iam/saml_identity_providers_fetch.go:2]: rm altogether?

File: plugins/source/aws/resources/services/iam/signing_certificates_fetch.go [hunk 3]
```diff
@@ -1,24 +1 @@
 package iam
-
```
Comment (candiduslynx) [inline plugins/source/aws/resources/services/iam/signing_certificates_fetch.go:2]: rm?

File: plugins/source/aws/resources/services/iam/users_fetch.go [hunk 4]
```diff
@@ -3,110 +3,12 @@ package iam
...
-			return err
-		}
-		if err := resource.Set("last_used_service_name", output.AccessKeyLastUsed.ServiceName); err != nil {
-			return err
-		}
-	}
-	return nil
-}
-
 func fetchIamUserAttachedPolicies(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
```
Comment (candiduslynx) [inline plugins/source/aws/resources/services/iam/users_fetch.go:12]: move this, too, and rm file?

File: plugins/source/aws/resources/services/iam/saml_identity_providers.go (hunk 1 above)
Comment (bbernays) [reply plugins/source/aws/resources/services/iam/saml_identity_providers.go:62]: I am not trying to fix or change the code, just trying to change the structure/formatting the files

**Review by candiduslynx — APPROVED** [review]

**Review by candiduslynx — APPROVED** [review]

**Pull request merged**